modules/challenge/src/main/ChallengeBulk.scala

```diff
@@ -96,6 +99,25 @@ final class ChallengeBulkApi(
   }
 
   private def makePairings(bulk: ScheduledBulk): Funit = {
+    val clock = bulk.clock.left.toOption.map(_.toClock)
+    def makeChess(variant: chess.variant.Variant): chess.Game =
+      chess.Game(situation = Situation(variant), clock = clock)
+
+    val baseState = bulk.fen.ifTrue(bulk.variant.fromPosition || bulk.variant.chess960) flatMap {
+      Forsyth.<<<@(bulk.variant, _)
+    }
+    val (chessGame, state) = baseState.fold(makeChess(bulk.variant) -> none[SituationPlus]) {
+      case sp @ SituationPlus(sit, _) =>
+        val game = chess.Game(
+          situation = sit,
+          turns = sp.turns,
+          startedAtTurn = sp.turns,
+          clock = clock
+        )
+        if (bulk.variant.fromPosition && Forsyth.>>(game).initial)
+          makeChess(chess.variant.Standard) -> none
+        else game                           -> baseState
+    }
```

Comment:
This code and the pipe stuff below is pretty much just a copy from [ChallengeJoiner](https://github.com/lichess-org/lila/blob/master/modules/challenge/src/main/ChallengeJoiner.scala#L35) so I guess it would make sense to refactor that into a function somewhere but not really sure where the best place would be 🤔